### ISSUE TITLE
Unable to rename the grok pattern, it lead extractor to remove from extractor list

### ISSUE THREAD

**Issue by rayeesnp** [Graylog2/graylog2-server#3602]
Added new grok patter in  graylog grok pattern list, and created extractor with this gork pater, renamed grok pattern name , unable to see the extractor from graylog to configure with updated name, also i am seeing exception in graylog server log 

## Expected Behavior

I should be able to update the extractor with the updated gork pattern name
## Current Behavior

After updating extractor name from grok pattern list,  extractor configured with gork pattern getting removed and unable to update extractor with latest name

The critical issue is, even after you fix the patter you can see messgaes are pushing to graylog, you need to restart the server to accept the messages 

## Steps to Reproduce (for bugs)

1.  Create grok pattern name "PALO_OUPUT"
2.  Created extractor "output_extractor" with grok pattern %{PALO_OUPUT}
3.  Update the grok pattern name from "PALO_OUPUT"  to "PALO_OUTPUT"
4.  Go to extractor page and select "output_extractor"

Result 

You cant see extractor "output_extractor" at step4, and seeing below error in log

`2017-03-10T12:50:08.811-05:00 ERROR [GrokExtractor] Unable to parse grok patterns
oi.thekraken.grok.api.exception.GrokException: Deep recursion pattern compilation of %{PALO_OUPUT}
	at oi.thekraken.grok.api.Grok.compile(Grok.java:356) ~[graylog.jar:?]
	at org.graylog2.inputs.extractors.GrokExtractor.<init>(GrokExtractor.java:79) [graylog.jar:?]
	at org.graylog2.inputs.extractors.ExtractorFactory.factory(ExtractorFactory.java:65) [graylog.jar:?]
	at org.graylog2.inputs.InputServiceImpl.getExtractors(InputServiceImpl.java:248) [graylog.jar:?]
	at org.graylog2.rest.resources.system.inputs.ExtractorsResource.list(ExtractorsResource.java:197) [graylog.jar:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_111]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_111]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_111]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_111]
	at org.glassfish.jersey.server.model.internal.ResourceMethodInvocationHandlerFactory$1.invoke(ResourceMethodInvocationHandlerFactory.java:81) [graylog.jar:?]
	at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher$1.run(AbstractJavaResourceMethodDispatcher.java:144) [graylog.jar:?]
	at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.invoke(AbstractJavaResourceMethodDispatcher.java:161) [graylog.jar:?]
	at org.glassfish.jersey.server.model.internal.JavaResourceMethodDispatcherProvider$TypeOutInvoker.doDispatch(JavaResourceMethodDispatcherProvider.java:205) [graylog.jar:?]
	at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.dispatch(AbstractJavaResourceMethodDispatcher.java:99) [graylog.jar:?]
	at org.glassfish.jersey.server.model.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:389) [graylog.jar:?]
	at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:347) [graylog.jar:?]
	at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:102) [graylog.jar:?]
	at org.glassfish.jersey.server.ServerRuntime$2.run(ServerRuntime.java:326) [graylog.jar:?]
	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:271) [graylog.jar:?]
	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:267) [graylog.jar:?]
	at org.glassfish.jersey.internal.Errors.process(Errors.java:315) [graylog.jar:?]
	at org.glassfish.jersey.internal.Errors.process(Errors.java:297) [graylog.jar:?]
	at org.glassfish.jersey.internal.Errors.process(Errors.java:267) [graylog.jar:?]
	at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:317) [graylog.jar:?]
	at org.glassfish.jersey.server.ServerRuntime.process(ServerRuntime.java:305) [graylog.jar:?]
	at org.glassfish.jersey.server.ApplicationHandler.handle(ApplicationHandler.java:1154) [graylog.jar:?]
	at org.glassfish.jersey.grizzly2.httpserver.GrizzlyHttpContainer.service(GrizzlyHttpContainer.java:384) [graylog.jar:?]
	at org.glassfish.grizzly.http.server.HttpHandler$1.run(HttpHandler.java:224) [graylog.jar:?]
	at com.codahale.metrics.InstrumentedExecutorService$InstrumentedRunnable.run(InstrumentedExecutorService.java:176) [graylog.jar:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [?:1.8.0_111]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [?:1.8.0_111]
	at java.lang.Thread.run(Thread.java:745) [?:1.8.0_111]
2017-03-10T12:50:08.815-05:00 ERROR [InputServiceImpl] Cannot build extractor from persisted data. Skipping.
org.graylog2.ConfigurationException: Unable to parse grok patterns
	at org.graylog2.inputs.extractors.GrokExtractor.<init>(GrokExtractor.java:82) ~[graylog.jar:?]
	at org.graylog2.inputs.extractors.ExtractorFactory.factory(ExtractorFactory.java:65) ~[graylog.jar:?]
	at org.graylog2.inputs.InputServiceImpl.getExtractors(InputServiceImpl.java:248) [graylog.jar:?]
	at org.graylog2.rest.resources.system.inputs.ExtractorsResource.list(ExtractorsResource.java:197) [graylog.jar:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_111]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_111]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_111]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_111]
	at org.glassfish.jersey.server.model.internal.ResourceMethodInvocationHandlerFactory$1.invoke(ResourceMethodInvocationHandlerFactory.java:81) [graylog.jar:?]
	at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher$1.run(AbstractJavaResourceMethodDispatcher.java:144) [graylog.jar:?]
	at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.invoke(AbstractJavaResourceMethodDispatcher.java:161) [graylog.jar:?]
	at org.glassfish.jersey.server.model.internal.JavaResourceMethodDispatcherProvider$TypeOutInvoker.doDispatch(JavaResourceMethodDispatcherProvider.java:205) [graylog.jar:?]
	at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.dispatch(AbstractJavaResourceMethodDispatcher.java:99) [graylog.jar:?]
	at org.glassfish.jersey.server.model.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:389) [graylog.jar:?]
	at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:347) [graylog.jar:?]
	at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:102) [graylog.jar:?]
	at org.glassfish.jersey.server.ServerRuntime$2.run(ServerRuntime.java:326) [graylog.jar:?]
	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:271) [graylog.jar:?]
	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:267) [graylog.jar:?]
	at org.glassfish.jersey.internal.Errors.process(Errors.java:315) [graylog.jar:?]
	at org.glassfish.jersey.internal.Errors.process(Errors.java:297) [graylog.jar:?]
	at org.glassfish.jersey.internal.Errors.process(Errors.java:267) [graylog.jar:?]
	at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:317) [graylog.jar:?]
	at org.glassfish.jersey.server.ServerRuntime.process(ServerRuntime.java:305) [graylog.jar:?]
	at org.glassfish.jersey.server.ApplicationHandler.handle(ApplicationHandler.java:1154) [graylog.jar:?]
	at org.glassfish.jersey.grizzly2.httpserver.GrizzlyHttpContainer.service(GrizzlyHttpContainer.java:384) [graylog.jar:?]
	at org.glassfish.grizzly.http.server.HttpHandler$1.run(HttpHandler.java:224) [graylog.jar:?]
	at com.codahale.metrics.InstrumentedExecutorService$InstrumentedRunnable.run(InstrumentedExecutorService.java:176) [graylog.jar:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [?:1.8.0_111]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [?:1.8.0_111]
	at java.lang.Thread.run(Thread.java:745) [?:1.8.0_111]
2017-03-10T12:50:35.540-05:00 ERROR [GrokExtractor] Unable to parse grok patterns`

## Your Environment

* Graylog Version:
* Elasticsearch Version: 2.2
* MongoDB Version:
* Operating System:
* Browser version:

**Comment by dennisoelkers** [comment]
I tried to reproduce the issue on `2.2.1` and was unable to do so. Editing the extractor referencing a nonexisting (renamed) grok pattern was no problem, only when clicking the "Try" button it returned an error (of course). Is there any additional information you can provide allowing us to reproduce it? Which exact version are you using?

**Comment by rayeesnp** [comment]
@dennisoelkers  I am using graylog 2.2.1, I am unable to reproduce this issue now,  can you resolve for now, If I am seeing this issue again, I wil repopen this ticket

**Comment by urban-moniker** [comment]
Hi,

May not be the best place to post this, but its seems related.

I have an issue - fairly minor but annoying - that I am getting the following entries logged.  I think this is because I renamed some lookup tables.  I have fairly much wiped them out and started again now - I removed the extractors that referenced them, and the input, and re-created from scratch, but I'm still getting these errors logged as if the old data is being held somewhere:-

2019-03-24T15:27:42.292Z ERROR [InputServiceImpl] Cannot build extractor from persisted data. Skipping.
java.lang.IllegalStateException: Configured lookup table <ad-user-created> doesn't exist
        at org.graylog2.inputs.extractors.LookupTableExtractor.<init>(LookupTableExtractor.java:57) ~[graylog.jar:?]

Note that I'm using GL 3.0.0.

Rgds
